Improve kerberos documents (#4771)
apache/pulsar: site2/website/versioned_docs/version-2.4.0/security-kerberos.md

    and the location of the keytab where the principal is stored. It allows the broker to use the keytab specified in this section.
2. `PulsarClient` is a section name in the JASS file used by each client. This section tells the client which principal to use inside Kerberos
    and the location of the keytab where the principal is stored. It allows the client to use the keytab specified in this section.

>     In the following example, this `PulsarClient` section will also be reused in both the Pulsar internal admin configuration and in cli command of `bin/pulsar-client`, `bin/pulsar-perf` and `bin/pulsar-admin`. You can also add different section for different use case.

Reviewer: ```suggestion
    In the following example, this `PulsarClient` section will also be reused in both the Pulsar internal admin configuration and in CLI command of `bin/pulsar-client`, `bin/pulsar-perf` and `bin/pulsar-admin`. You can also add different sections for different use cases.
```

You can have 2 separate JAAS configuration files: 

> * the file for broker will have sections of both `PulsarBroker` and `PulsarClient`; 

Reviewer: ```suggestion
* the file for a broker has sections of both `PulsarBroker` and `PulsarClient`; 
```

> * the file for client only have `PulsarClient` section.

Reviewer: ```suggestion
* the file for a client only has a `PulsarClient` section.
```

### Kerberos configuration for Brokers

#### Configure `broker.conf` file
 
 In the `broker.conf` file, set Kerberos related configurations.

 - Set `authenticationEnabled` to `true`;
 - Set `authenticationProviders` to choose `AuthenticationProviderSasl`;
 - Set `saslJaasClientAllowedIds` regex for principal that is allowed to connect to broker;
 - Set `saslJaasBrokerSectionName` that corresponding to the section in JAAS configuration file for broker;
 
 To make Pulsar internal admin client work properly, you need to set the configuration in the `broker.conf` file as below: 
 - Set `brokerClientAuthenticationPlugin` to client plugin `AuthenticationSasl`;

>  - Set `brokerClientAuthenticationParameters` to value in Json string `{"saslJaasClientSectionName":"PulsarClient", "serverType":"broker"}`, in which `PulsarClient` is the section name in above `pulsar_jaas.conf` file, and `"serverType":"broker"` indicate that internal admin client will connect to a Pulsar Broker;

Reviewer: ```suggestion
 - Set `brokerClientAuthenticationParameters` to value in JSON string `{"saslJaasClientSectionName":"PulsarClient", "serverType":"broker"}`, in which `PulsarClient` is the section name in above `pulsar_jaas.conf` file, and `"serverType":"broker"` indicate that internal admin client will connect to a Pulsar Broker;
```

 Here is an example:

```
authenticationEnabled=true
authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderSasl
saslJaasClientAllowedIds=.*client.*
saslJaasBrokerSectionName=PulsarBroker

> ## Authentication settings of the broker itself. Used when the broker connects to other brokers

Reviewer: The heading (##) is too long?

Author: This not head, it is comments in shell

Make sure that the keytabs configured in the `pulsar_jaas.conf` file and kdc server in the `krb5.conf` file are reachable by the operating system user who is starting pulsar client.

> #### configure CLI tools

Reviewer: ```suggestion
#### Configure CLI tools
```

If you are using a command-line tool (such as `bin/pulsar-client`, `bin/pulsar-perf` and `bin/pulsar-admin`), you need to preform the following steps:

Step 1. Config your `client.conf`.
```shell
authPlugin=org.apache.pulsar.client.impl.auth.AuthenticationSasl
authParams={"saslJaasClientSectionName":"PulsarClient", "serverType":"broker"}
```

> Step 2.  Set JVM parameters for JAAS configuration file and krb5 configuration file with additional options.

Reviewer: ```suggestion
Step 2. Set JVM parameters for JAAS configuration file and krb5 configuration file with additional options.
```

```shell
   -Djava.security.auth.login.config=/etc/pulsar/pulsar_jaas.conf -Djava.security.krb5.conf=/etc/pulsar/krb5.conf 
```

You can add this at the end of `PULSAR_EXTRA_OPTS` in the file [`pulsar_tools_env.sh`](https://github.com/apache/pulsar/blob/master/conf/pulsar_tools_env.sh),

> Or add this line `OPTS="$OPTS -Djava.security.auth.login.config=/etc/pulsar/pulsar_jaas.conf -Djava.security.krb5.conf=/etc/pulsar/krb5.conf "` directly in to the CLI tool script.

Reviewer: ```suggestion
or add this line `OPTS="$OPTS -Djava.security.auth.login.config=/etc/pulsar/pulsar_jaas.conf -Djava.security.krb5.conf=/etc/pulsar/krb5.conf "` directly to the CLI tool script.
```

```
authProvider.1=org.apache.zookeeper.server.auth.SASLAuthenticationProvider
requireClientAuthScheme=sasl
```

And add a section of `Client` configurations in the file `pulsar_jaas.conf`, which is used by Pulsar Broker:

```
 Client {
   com.sun.security.auth.module.Krb5LoginModule required
   useKeyTab=true
   storeKey=true
   useTicketCache=false
   keyTab="/etc/security/keytabs/pulsarbroker.keytab"
   principal="broker/localhost@EXAMPLE.COM";
};
```

In this setting, Pulsar Broker's principal and keyTab file indicates Broker's role when authenticating with ZooKeeper.

## Regarding authentication between BookKeeper and Broker

Pulsar Broker acts as a Kerberos client when authenticating with Bookie. According to [BookKeeper document](http://bookkeeper.apache.org/docs/latest/security/sasl/), you need to add `bookkeeperClientAuthenticationPlugin` parameter in `broker.conf`:

```
bookkeeperClientAuthenticationPlugin=org.apache.bookkeeper.sasl.SASLClientProviderFactory
```

> In this setting, `SASLClientProviderFactory` will create a BookKeeper SASL client in Broker, and Broker use created SASL client to authenticate with a Bookie node.

Reviewer: ```suggestion
In this setting, `SASLClientProviderFactory` creates a BookKeeper SASL client in a broker, and the broker uses the created SASL client to authenticate with a bookie node.
```
